pkg/gui/presentation/branches_test.go

```diff
@@ -0,0 +1,214 @@
+package presentation
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jesseduffield/lazygit/pkg/commands/models"
+	"github.com/jesseduffield/lazygit/pkg/gui/presentation/icons"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"github.com/jesseduffield/lazygit/pkg/utils"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getBranchDisplayStrings(t *testing.T) {
+	scenarios := []struct {
+		branch               *models.Branch
+		itemOperation        types.ItemOperation
+		fullDescription      bool
+		viewWidth            int
+		useIcons             bool
+		checkedOutByWorktree bool
+		expected             []string
+	}{
+		// First some tests for when the view is wide enough so that everything fits:
+		{
+			branch:               &models.Branch{Name: "branch_name", Recency: "1m"},
+			itemOperation:        types.ItemOperationNone,
+			fullDescription:      false,
+			viewWidth:            100,
+			useIcons:             false,
+			checkedOutByWorktree: false,
+			expected:             []string{"1m", "branch_name"},
+		},
+		{
+			branch:               &models.Branch{Name: "branch_name", Recency: "1m"},
+			itemOperation:        types.ItemOperationNone,
+			fullDescription:      false,
+			viewWidth:            100,
+			useIcons:             false,
+			checkedOutByWorktree: true,
+			expected:             []string{"1m", "branch_name (worktree)"},
+		},
+		{
+			branch:               &models.Branch{Name: "branch_name", Recency: "1m"},
+			itemOperation:        types.ItemOperationNone,
+			fullDescription:      false,
+			viewWidth:            100,
+			useIcons:             true,
+			checkedOutByWorktree: true,
+			expected:             []string{"1m", "󰘬", "branch_name 󰌹"},
```

Comment:
I'm using literal nerd font characters here, but they don't show up on github of course. Still, I like how the tests read when looking at them in an editor that uses a nerd font, so I'm inclined to keep it this way.

Reply:
Added a paragraph in 41bb6b96d53b.